Sources/GryphonLib/KotlinTranslator.swift

```diff
@@ -730,16 +741,43 @@ public class KotlinTranslator {
 		else {
 			result.append(try translateSubtrees(
 				statements,
-				withIndentation: indentation,
+				withIndentation: !singleExpressionStatement ? indentation : "",
```

Comment:
This could cause problems if the single expression gets broken up into multiple lines (e.g. a closure or a long function call), since its indentation would have been reset. For instance, try translating this:

```` swift
class A {
	class B {
		class C {
			func aaaaaaaaaaaaa(bbbbbbbbbbbbbbbb: Int, ccccccccccccc: Int, ddddddddddddddddddddddddd: Int, eeeeeeeeeee: Int) -> String {
				"abc"
			}
			
			func d() -> String {
				aaaaaaaaaaaaa(bbbbbbbbbbbbbbbb: 0, ccccccccccccc: 0, ddddddddddddddddddddddddd: 0, eeeeeeeeeee: 0)
			}
		}
	}
}
````

It's a silly example, but it demonstrates the two problems I was talking about.

You could solve this by making `isSingleExpressionFunction` return the single expression itself (or nil). Then you could check if the expression is nil when adding the `{`s, and use `translateExpression` on it here with the correct indentation.

Reply:
You really know your stuff, thanks for that suggestion.
This looks way nicer now!

```kotlin
internal open class A {
    open class B {
        open class C {
            open fun aaaaaaaaaaaaa(
                bbbbbbbbbbbbbbbb: Int,
                ccccccccccccc: Int,
                ddddddddddddddddddddddddd: Int,
                eeeeeeeeeee: Int)
                : String
                = "abc"

            open fun d(): String = aaaaaaaaaaaaa(
                    bbbbbbbbbbbbbbbb = 0,
                    ccccccccccccc = 0,
                    ddddddddddddddddddddddddd = 0,
                    eeeeeeeeeee = 0)
        }
    }
}
```